bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVConsoleHandler.java

```diff
@@ -0,0 +1,111 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Console commands for interacting with the Tesla integration
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = ConsoleCommandExtension.class, immediate = true)
+public class MagentaTVConsoleHandler extends AbstractConsoleCommandExtension {
+
+    private static final String CMD_LOGIN = "login";
+
+    private final Logger logger = LoggerFactory.getLogger(MagentaTVConsoleHandler.class);
+    private final MagentaTVOAuth oauth = new MagentaTVOAuth();
+
+    @Reference(cardinality = ReferenceCardinality.OPTIONAL)
+    private @Nullable ClientBuilder injectedClientBuilder;
+
+    @Activate
+    public MagentaTVConsoleHandler() {
+        super(BINDING_ID, "Interact with the " + BINDING_ID + " integration.");
+    }
+
+    @Override
+    public void execute(String[] args, Console console) {
+        if (args.length > 0) {
+            String subCommand = args[0];
+            switch (subCommand) {
+                case CMD_LOGIN:
+                    if (args.length == 1) {
+                        try {
+                            BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+                            console.print("Login Name (email): ");
+                            String username = br.readLine();
+                            console.print("Password: ");
+                            String pwd = br.readLine();
+                            console.println("Attempting login...");
+                            login(console, username, pwd);
+                        } catch (IOException e) {
+                            console.println(e.toString());
+                        }
+                    } else if (args.length == 3) {
+                        login(console, args[1], args[2]);
+                    } else {
+                        printUsage(console);
+                    }
+                    break;
+
+                default:
+                    console.println("Unknown command '" + subCommand + "'");
+                    printUsage(console);
+                    break;
+            }
+        }
+    }
+
+    @Override
+    public List<String> getUsages() {
+        return Arrays.asList(buildCommandUsage(CMD_LOGIN + " [<login email>] [<password>]",
+                "Authenticates the login name and provides the technical User Id (userId parameter)."));
+    }
+
+    private void login(Console console, String username, String password) {
+        try {
+            logger.info("Performing OAuth for user {}", username);
+            String userId = oauth.getUserId(username, password);
+            console.println("Login successful, returned userId is " + userId);
+            console.println("Edit thing configuration and copy this value to the field User Id (userId)");
+            logger.info("User Id for user {} is {}", username, userId);
```

Comment:
just noted. The UI doesn't show a field "User ID" and the user is not aware that this parameter is named userId. So it doesn't make sense to talk about "parameter userId". From the user's point of view is also "Account Name" and not accountName. Therefore I'll use "User ID" for user doc/messages and userId for technical messages.

Reply:
> it doesn't make sense to talk about "parameter userId".

It makes a lot of sense, because this is exactly what is [documented as parameters](https://github.com/openhab/openhab-addons/pull/7864/files#diff-5857524926f49c884a3111d0844a2caaR66).

Also note that the console command is meant for people doing textual configuration and the `userId` parameter is exactly what they are after.

Reply:
For me the console command is for users who doesn't want to store credentials in the configuration.
How does a UI user understands that userId means the field User ID, which is the description for the parameter userId? This is not visible in the UI

Changed the message to:
Login successful, returned User ID is xxxx
Edit thing configuration and copy this value to the field User ID or use it as parameter userId for the textual configuration.

ok?

Reply:
ok.